src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -1004,9 +1020,9 @@ public BoundStatement SequencePointWithSpan(CSharpSyntaxNode syntax, TextSpan sp
             return new BoundSequencePointWithSpan(syntax, statement, span);
         }
 
-        public BoundStatement HiddenSequencePoint(BoundStatement statementOpt = null)
+        public BoundStatement HiddenSequencePoint(BoundStatement? statementOpt = null)
         {
-            return new BoundSequencePoint(null, statementOpt) { WasCompilerGenerated = true };
+            return new BoundSequencePoint(null!, statementOpt) { WasCompilerGenerated = true };
```

Comment:
>null! [](start = 42, length = 5)

May be worth a comment.
Why not mark the constructor parameter as nullable?

Reply:
Because it is passed on to the base constructor which declares that it does not accept null.  I cannot put a ! in the generated bound node code to work around that.  This is a one-off exception to the requirement that the type be non-null.

---
In reply to: [386590295](https://github.com/dotnet/roslyn/pull/42068#discussion_r386590295) [](ancestors = 386590295)